openff/interchange/components/potentials.py

```diff
@@ -127,9 +127,12 @@ def validate_key_map(v: Any, handler, info) -> dict:
                     key_class = ImproperTorsionKey
                 case "LibraryCharges":
                     key_class = LibraryChargeTopologyKey  # type: ignore[assignment]
-                case "ToolkitAM1BCCHandler":
-                    key_class = SingleAtomChargeTopologyKey  # type: ignore[assignment]
-                case "molecules_with_preset_charges":
+                case (
+                    "ToolkitAM1BCCHandler"
+                    | "molecules_with_preset_charges"
+                    | "NAGLChargesHandler"
+                    | "ChargeIncrementModelHandler"
+                ):
```

Comment:
This (Python 3.10+) or some variant of it should be all that's needed to port this into older versions ... I hope. The comparison are between strings, which hopefully provides some flexibility compared to comparing types or classes